File: RedSheet.DbEntities/Models/vLiteTheirTeamMemberRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLiteTheirTeamMemberRecords", Schema = "dbo")]
	[Serializable]
	public partial class vLiteTheirTeamMemberRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int LiteProjectBackgroundId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int LiteTheirTeamMemberId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string LiteTheirTeamMemberName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int PersonalityId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string Position { get; set; } 

	public vLiteTheirTeamMemberRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-lite-their-team-member-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLiteTheirTeamMemberRecord {
    constructor(vLiteTheirTeamMemberRecord?: vLiteTheirTeamMemberRecord )  {
        let properties = [ "liteProjectBackgroundId", "liteTheirTeamMemberId", "liteTheirTeamMemberName", "personalityId", "position",];
        for (let property of properties)
            if (vLiteTheirTeamMemberRecord && vLiteTheirTeamMemberRecord[property])
                this[property] = vLiteTheirTeamMemberRecord[property];
    }
 
	liteProjectBackgroundId : number =   0 ;
 
    @range(1,2147483647)
	liteTheirTeamMemberId : number =   undefined;
 
    @required()
    @maxLength(500)
	liteTheirTeamMemberName : string =   undefined;
 
    @range(1,2147483647)
	personalityId : number =   undefined;
 
    @required()
    @maxLength(500)
	position : string =   undefined;


}
